base/twiceprecision.jl

```diff
@@ -198,6 +300,8 @@ struct TwicePrecision{T}
     lo::T    # least significant bits
 end
 
+(::Type{<:Tuple})(t::TwicePrecision) = (t.hi, t.lo)
```

Comment:
Why is this conversion necessary? Can't the algorithms in this PR work with `TwicePrecision` directly?

This definition is also REALLY wide, as it allows things like this:

```
julia> (::Type{<:Tuple})(t::Base.TwicePrecision) = (t.hi, t.lo)

julia> Tuple{Int, Int}(Base.TwicePrecision(0.0, 1.0))
(0.0, 1.0)

julia> Tuple{String, Any}(Base.TwicePrecision(0.0, 1.0))
(0.0, 1.0)
```

Which I don't think is the intention.

Reply:
> Why is this conversion necessary? Can't the algorithms in this PR work with TwicePrecision directly?

You'll see why when I push the changes I have locally.

> This definition is also REALLY wide

OK, will change it to `Tuple(t::Base.TwicePrecision) = (t.hi, t.lo)`, I think that should be OK.